### PR TITLE
doc: opt animator doc

### PR DESCRIPTION
**Pull request by luzhuang**
### Please check if the PR fulfills these requirements

- [x] The commit message follows our [guidelines](https://github.com/galacean/engine/blob/main/.github/COMMIT_MESSAGE_CONVENTION.md)
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

## Summary by CodeRabbit

- **New Features**
  - Introduced comprehensive documentation for Animator Controller, Animator State Machine, and AnimatorControllerLayer, enhancing user understanding of animation management.
  - Added a new section in Animator documentation for improved clarity and logical flow of instructions.

- **Documentation**
  - Updated and refined existing documentation for Animator components, including clearer editor and script usage instructions.
  - Expanded details on animation playback control, blending modes, and state transitions for better user implementation.
  - Corrected terminology in the modelMesh documentation to enhance clarity regarding BlendShape animation effects.

- **Chores**
  - Adjusted the ordering of several documents within the animation documentation structure for improved navigation.

### CODE REVIEW

File: docs/zh/animation/animator.mdx [hunk 1]
```diff
@@ -15,23 +16,19 @@ label: Animation
 
 ## 编辑器使用
 
-1. 当我们把模型拖入到场景中，模型以初始姿态展示出来，但是并不会播放任何动画，我们需要在模型实体上添加动画控制组件（[Animator](/apis/core/#Animator)）
+1. 当我们把模型拖入到场景中，模型以初始姿态展示出来，但是并不会播放任何动画，我们需要在模型实体上找到动画控制组件（[Animator](/apis/core/#Animator)）并为它绑定一个 [动画控制器](/docs/animation-animatorController) 资产
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/animation/animator.mdx:19]: 当我们把模型拖入到场景中，模型将以初始姿态展示出来，并不会播放任何动画。

File: docs/zh/animation/animator.mdx [hunk 2]
```diff
@@ -15,23 +16,19 @@ label: Animation
...
-2. 动画控制组件（[Animator](/apis/core/#Animator)）需要绑定一个 [动画控制器](/docs/animation-animatorController) 资产，我们创建并绑定
-
-![3](https://mdn.alipayobjects.com/huamei_3zduhr/afts/img/A*irT7SZvw4N8AAAAAAAAAAAAADsJ_AQ/original)
-
-![4](https://mdn.alipayobjects.com/huamei_3zduhr/afts/img/A*VtX3RJR8kdMAAAAAAAAAAAAADsJ_AQ/original)
-
-3. 至此你在导出的项目中就可以通过 `animator.play` 播放[动画控制器](/docs/animation-animatorController)中的动画了。
-
-如果你没有为实体添加 动画控制组件（[Animator](/apis/core/#Animator)）的话 Galacean Engine 会为你默认创建一个并且 [动画控制器](/docs/animation-animatorController) 中默认添加了模型的所有动画片段，当然除此以外你可以通过 [动画控制器](/docs/animation-animatorController) 实现更多的功能。
+2. 我们创建一个[动画控制器](/docs/animation-animatorController) 资产并绑定到模型上
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/animation/animator.mdx:22]: 去掉「我们」

File: docs/zh/animation/animator.mdx [hunk 3]
```diff
@@ -15,23 +16,19 @@ label: Animation
...
+![alt text](https://mdn.alipayobjects.com/huamei_3zduhr/afts/img/A*PxHwTrU58yAAAAAAAAAAAAAADsJ_AQ/original)
+![alt text](https://mdn.alipayobjects.com/huamei_3zduhr/afts/img/A*Ds5TTosYiDYAAAAAAAAAAAAADsJ_AQ/original)
+3. 编辑动画控制器（[详见](/docs/animation-animatorController)）后你就可以按照动画控制器的逻辑播放其中的动画了。
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/animation/animator.mdx:25]: [动画控制器](/docs/animation-animatorController)

File: docs/zh/animation/animatorController.mdx [hunk 4]
```diff
@@ -0,0 +1,59 @@
+---
+order: 2
+title: 动画控制器
+type: 动画
+label: Animation
+---
+动画控制器（[AnimatorController](/apis/core/#AnimatorController)）可以帮助你为角色或其他动画对象组织一组动画的播放逻辑。它通过 动画状态（[AnimatorState](/apis/core/#AnimatorState)）引用了[动画片段](/docs/animation-clip)（[AnimationClip](/apis/core/#AnimationClip)）在其中使用，并通过[动画状态机](/docs/state-machine)（[AnimatorStateMachine](/apis/core/#AnimatorStateMachine)）像流程图一样可视化的组织动画对象的动画播放逻辑
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/animation/animatorController.mdx:7]: 动画控制器（[AnimatorController](/apis/core/#AnimatorController)）为目标对象组织一系列动画状态的播放逻辑。它通过动画状态机，像流程图一样可视化地控制不同动画状态的切换条件和时机，并播放状态引用的动画片段。

File: docs/zh/animation/state-machine.mdx [hunk 5]
```diff
@@ -0,0 +1,274 @@
+---
+order: 4
+title: 动画状态机
+type: 动画
+label: Animation
+---
+## 简介
+
+动画状态机（[AnimatorStateMachine](/apis/core/#AnimatorStateMachine)）是一种用于控制和管理动画切换的工具，你可以用它添加各种动画状态及其之间的转换规则，使角色或动画对象能够在不同动作之间自然的切换。
```
Comment (cptbtptpbcptdtptp) [inline docs/zh/animation/state-machine.mdx:9]: 感觉白话描述状态机做的三件事更好理解，这篇文档也主要在讲这个内容：
- 什么时机切换状态
- 状态之间怎么过渡
- 状态怎么播放（动画状态的属性）

File: docs/zh/animation/animator.mdx (hunk 2 above)
Comment (luzhuang) [reply docs/zh/animation/animator.mdx]: done

File: docs/zh/animation/animator.mdx (hunk 3 above)
Comment (luzhuang) [reply docs/zh/animation/animator.mdx]: done

File: docs/zh/animation/state-machine.mdx (hunk 5 above)
Comment (luzhuang) [reply docs/zh/animation/state-machine.mdx:9]: 这里是简介主要介绍它对用户有啥用，先不涉及各种概念

File: docs/zh/animation/animatorController.mdx (hunk 4 above)
Comment (luzhuang) [reply docs/zh/animation/animatorController.mdx]: done

**Review by cptbtptpbcptdtptp — APPROVED** [review]
+1

**Pull request merged**